docker-compose.ci.yml

```diff
@@ -0,0 +1,17 @@
+version: '2.4'
+
+services:
+  cibuild:
+    image: adoptopenjdk/openjdk8:slim
+    environment:
+      - CI=${CI:-}
+    volumes:
+      - ~/.gradle/caches/modules-2:/root/.gradle/caches/modules-2:z
+      - ~/.gradle/caches/forge_gradle:/root/.gradle/caches/forge_gradle:z
+      - ~/.gradle/wrapper/dists:/root/.gradle/wrapper/dists:z
+      - ./:/opt/build:Z
+    working_dir: /opt/build
+    entrypoint: bash -c 'set -x
+      && chmod +x gradlew
+      && ./gradlew --no-daemon -p buildSrc clean
+      && ./gradlew --no-daemon --no-build-cache clean build'
```

Comment:
I'd make `gradlew` executable and then commit that to the repository. Also, consider making the entrypoint `./gradlew` itself, and adjusting `cibuild` to look like:

```console
docker-compose -f docker-compose.ci.yml \
  run --rm gradle \
  :buildSrc:clean clean build
```

See: https://docs.gradle.org/current/userguide/intro_multi_project_builds.html#sec:executing_tasks_by_fully_qualified_name

---

The Gradle docs say:

> Since Gradle 3.0, we enable Daemon by default and recommend using it for both developers' machines and Continuous Integration servers. However, if you suspect that Daemon makes your CI builds unstable, you can disable it to use a fresh runtime for each build since the runtime is completely isolated from any previous builds.

Was there a specific reason you disabled the daemon and build cache?

See: https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:disabling_the_daemon

Reply:
> I'd make gradlew executable and then commit that to the repository.

I believe `gradlew` already is executable in the project. It is just common to add executable perms to it just to be safe.

> Also, consider making the entrypoint ./gradlew itself, and adjusting cibuild to look like

`:buildSrc` is not a subproject. It is a special directory for custom gradle plugins. `./gradlew :buildSrc:clean` will fail and is not an equivalent to doing `./gradlew -p buildSrc clean`.  It is for that reason I don't just make `gradlew` the entrypoint